core/src/main/java/org/elasticsearch/common/util/AbstractArray.java

```diff
@@ -28,11 +29,12 @@
 
 abstract class AbstractArray implements BigArray {
 
+    @Nullable
     private final BigArrays bigArrays;
     public final boolean clearOnResize;
     private final AtomicBoolean closed = new AtomicBoolean(false);
 
-    AbstractArray(BigArrays bigArrays, boolean clearOnResize) {
+    AbstractArray(@Nullable BigArrays bigArrays, boolean clearOnResize) {
```

Comment:
@jpountz I completely missed the `NON_RECYCLING_INSTANCE`!  It works perfectly.  I made the change.